pkg/trace/propagation.go

```diff
@@ -111,20 +78,6 @@ func (t *otelTracer) toHeaders(ctx context.Context) map[string][]string {
 	propagator := otel.GetTextMapPropagator()
 	propagator.Inject(ctx, otelpropagation.HeaderCarrier(result))
 
-	// Honeycomb expects
-	headers := map[string]string{
-		strings.ToLower(OtelPropagationHeader): result.Get(OtelPropagationHeader),
-	}
-
-	_, prop, err := propagation.UnmarshalW3CTraceContext(ctx, headers)
-	if err != nil {
-		return result
-	}
-
-	result.Set(
-		propagation.TracePropagationHTTPHeader,
-		propagation.MarshalHoneycombTraceContext(prop))
-
```

Comment:
As far as I can tell, this block of code:
- Serializes the current trace context to the W3C header format.
- Uses honeycomb beeline code to parse those headers into structures the beeline libs understand.
- Re-serializes this beeline-formatted data into Honeycomb-style headers.
- Sets those Honeycomb headers in the return value.

In other words, it's a hack to ensure we forward HC-style as well as OpenTelemetry-style headers.  This seems unnecessary if we're dropping support for receiving those headers, and we think that pretty much everyone has been ignoring the HC-style headers for quite some time now.  Hence this deletion.